apps/designer/app/designer/features/style-panel/style-section.tsx

```diff
@@ -0,0 +1,777 @@
+                      },
+                    }),
+                  }}
+                >
+                  <Icon />
+                </IconButton>
+              </Tooltip>
+              <TextField
+                type="number"
+                defaultValue={parseFloat(String(value.value)) || 0}
+                // @todo looses input state and focus
+                // onChange={(event: ChangeEvent<HTMLInputElement>) => setValue(event.target.value + "px")
+                onKeyDown={(event: KeyboardEvent<HTMLInputElement>) =>
+                  event.key === "Enter" &&
+                  setValue(event.currentTarget.value + "px")
+                }
+                css={{
+                  fontWeight: "500",
+                  paddingLeft: "calc($5 + 6px)",
+                  gridArea: "1 / 1 / -1 / -1",
+                  cursor: "default",
+                }}
+              />
+            </Grid>
+          );
+        }
+      }
+    }
+  }
+
+  return (
+    <Grid columns={2} align="center" gapX="1">
+      {/* @todo needs icon variant */}
+      <PropertyName property={styleConfig.property} label={styleConfig.label} />
+      <Flex align="center" css={{ gridColumn: "2/4" }} gap="1">
+        <Combobox
+          label={styleConfig.property}
+          items={styleConfig.items}
+          // state={value.type === "invalid" ? "invalid" : undefined}
+          value={String(value.value)}
+          // @todo new combobox doesn't include any of these event handlers
+          // onValueSelect={setValue}
+          // onValueEnter={setValue}
+          // onItemEnter={(value: unknown) => {
+          //   setValue(value as string, { isEphemeral: true });
+          // }}
+          // onItemLeave={() => {
+          //   setValue(String(value.value), { isEphemeral: true });
+          // }}
+        />
+        <Unit value={value} />
+      </Flex>
+    </Grid>
+  );
+};
+
+const SelectControl = ({
+  currentStyle,
+  inheritedStyle,
+  setProperty,
+  styleConfig,
+}: ControlProps) => {
+  // @todo show which instance we inherited the value from
+  const value = getFinalValue({
+    currentStyle,
+    inheritedStyle,
+    property: styleConfig.property,
+  });
+
+  if (value === undefined) return null;
+
+  const setValue = setProperty(styleConfig.property);
+
+  return (
+    <Flex align="center">
+      <PropertyName
+        property={styleConfig.property}
+        label={styleConfig.label}
+        css={{
+          fontWeight: "500",
+          marginRight: "$sizes$1",
+        }}
+      />
+      <Select
+        options={styleConfig.items.map(({ label }) => label)}
+        value={String(value.value)}
+        onChange={setValue}
+        ghost
+        css={{
+          gap: "calc($sizes$1 / 2)",
+          px: "$sizes$1",
+          fontWeight: "500",
+          textTransform: "capitalize",
+          "&:hover": { background: "none" },
+        }}
+      />
+    </Flex>
+  );
+};
+
+const IconButtonWithMenuControl = ({
```

Comment:
this could be a module in style-panel/lib, same like color picker and spacing

Reply:
sorry. I meant IconButtonWithMenu and its already there

Reply:
btw I just noticed you renamed lib to components, was the problem with lib infra related? I recall there was something ... maybe we should fix the lib to work as expected, because components folder is really meaningless, a lot of things are components, lib was specifically saying that those were things not coupled with the application state etc, so they are reusable even though still a local lib means they are designed for this feature specifically

Reply:
oh yeah, we should probably just use dist in others and stop ignoring lib

Reply:
its kindof wrong to use lib as a dist/build anyways

Reply:
can you send a separate pr for lib> dist renaming?